### PR TITLE
Feature/type script be

### PR DESCRIPTION
**Pull request by chinook25**
Typed as much functions as possible, replaced all backend js files with ts version.

### CODE REVIEW

File: node-backend/inProgressHandler.ts [hunk 1]
```diff
@@ -37,10 +39,9 @@ export default function InProgressHandler(db: any) {
     response: Response,
     next: () => {}
   ): void {
-    const user: {id: string} = getUser(request);
     const emptyInProgress = buildEmptyInProgress();
     inProgressWorkspaceRepository.create(
-      user.id,
+      getUserId(request).id,
```
Comment (DanielReid) [inline node-backend/inProgressHandler.ts:44]: renaming mistake?

File: node-backend/inProgressRepositoryService.ts [hunk 2]
```diff
@@ -60,15 +60,17 @@ export function mapWorkspace(
...
-      return {
-        id: queryCriterion.id,
-        title: queryCriterion.title,
-        description: queryCriterion.description,
-        isFavourable: queryCriterion.isfavourable,
-        dataSources: []
-      };
-    })
+    .map(
+      (queryCriterion): ICriterion => {
```
Comment (DanielReid) [inline node-backend/inProgressRepositoryService.ts:64]: `queryCriterion` isn't typed; might be unnecesary due to lodash generics? But looks odd to me

File: node-backend/inProgressRepositoryService.ts [hunk 3]
```diff
@@ -77,12 +79,14 @@ export function mapAlternatives(
...
   return _(alternatives)
     .sortBy('orderindex')
-    .map((queryAlternative) => {
-      return {
-        id: queryAlternative.id,
-        title: queryAlternative.title
-      };
-    })
+    .map(
+      (queryAlternative): IAlternative => {
```
Comment (DanielReid) [inline node-backend/inProgressRepositoryService.ts:83]: again missing type?

File: node-backend/orderingHandler.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+'use strict';
```
Comment (DanielReid) [inline node-backend/orderingHandler.ts:1]: These can be removed (JS only hint)

File: node-backend/util.ts [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
 import _ from 'lodash';
-import logger from './loggerTS';
-import {Request} from 'express';
-import { Error } from '@shared/interface/IError';
+import logger from './logger';
 
-export function getUser(request: Request) {
-  if (request.user) {
+export function getUserId(request: Request) {
+  if (request.user && request.user) {
```
Comment (DanielReid) [inline node-backend/util.ts:9]: ??

File: node-backend/util.ts [hunk 6]
```diff
@@ -22,21 +23,21 @@ export function handleError(error: Error, next: any): void {
   });
 }
 
-export function getRanges(problem: IProblem) {
+export function getRanges(problem: IProblem): Record<string, any> {
```
Comment (DanielReid) [inline node-backend/util.ts:26]: isn't there a better type of pvf range than `any`?

File: test/util.test.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
+'use strict';
+import {
+  getRanges,
+  getUserId,
+  handleError,
+  reduceProblem
+} from '../node-backend/util';
+
+describe('The utility', () => {
+  describe('getUser', () => {
```
Comment (DanielReid) [inline test/util.test.ts:10]: didn't get renamed here though :P

File: test/util.test.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        },
+        prefs: problem.prefs
+      };
+      expect(expectedResult).toEqual(result);
+    });
+  });
+
+  describe('getRanges', () => {
+    it('should return the scales ranges from a problem', () => {
+      const problem: any = {
```
Comment (DanielReid) [inline test/util.test.ts:74]: no types in this test?

File: node-backend/inProgressRepositoryService.ts (hunk 2 above)
Comment (chinook25) [reply node-backend/inProgressRepositoryService.ts:64]: Yes, but typed it anyway now :)

File: test/util.test.ts (hunk 8 above)
Comment (chinook25) [reply test/util.test.ts:74]: Didn't want to write a full blown IProblem when the test only uses a part of it.

File: node-backend/inProgressRepositoryService.ts [hunk 9]
```diff
@@ -80,7 +80,7 @@ export function mapAlternatives(
   return _(alternatives)
     .sortBy('orderindex')
     .map(
-      (queryAlternative): IAlternative => {
+      (queryAlternative: IAlternativeQueryResult): IAlternative => {
```
Comment (DanielReid) [inline node-backend/inProgressRepositoryService.ts:83]: forgot to tag comment with 'general for this file'

File: node-backend/util.ts [hunk 10]
```diff
@@ -26,7 +30,11 @@ export function handleError(error: Error, next: any): void {
 export function getRanges(problem: IProblem): Record<string, any> {
   return _.reduce(
     problem.criteria,
-    (accum: Record<string, any>, criterion, key): Record<string, any> => {
+    (
+      accum: Record<string, any>,
```
Comment (DanielReid) [inline node-backend/util.ts:34]: missed the accum

**Pull request merged**